src/internal.c

```diff
@@ -14789,31 +14789,6 @@ int SendData(WOLFSSL* ssl, const void* data, int sz)
         }
     }
 
-    /* last time system socket output buffer was full, try again to send */
-    if (ssl->buffers.outputBuffer.length > 0) {
```

Comment:
This code shouldn't be removed.
Instead add groupMsgs variable. Default is false and is set to true when sending early data.
Then check the variable here to see whether to send data in outputBuffer or to group with early data.